File: reposilite-backend/src/test/java/org/panda_lang/reposilite/utils/entity/crud/TestEntityController.java
package org.panda_lang.reposilite.utils.entity.crud;

import org.bson.types.ObjectId;
import org.panda_lang.reposilite.utils.entity.AbstractCrudController;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/tests")
public class TestEntityController extends AbstractCrudController<TestEntity, ObjectId, TestDto, TestDto> {

  protected TestEntityController(@Autowired TestCrudService service) {
    super(service);
  }

}

File: hub-backend/src/test/old_tests/org/panda_lang/hub/utils/entity/crud/TestEntityController.kt
/*
 * Copyright (c) 2020 Hub Team of panda-lang organization
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.panda_lang.hub.utils.entity.crud

import org.bson.types.ObjectId
import org.panda_lang.hub.utils.entity.AbstractCrudController
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import kotlin.Throws

@RestController
@RequestMapping("/api/tests")
class TestEntityController @Autowired protected constructor(service: TestCrudService?) : AbstractCrudController<TestCrudService?, TestEntity?, ObjectId?, TestDto?, TestDto?>(service)